numpy/core/src/multiarray/compiled_base.c

```
@@ -80,6 +80,349 @@ minmax(const npy_intp *data, npy_intp data_len, npy_intp *mn, npy_intp *mx)
     *mx = max;
 }
 
+NPY_NO_EXPORT PyObject *
+arr_histogram_uniform(PyObject *NPY_UNUSED(self), PyObject *args, PyObject *kwds)
+{
+    PyObject *obj_x, *obj_edges, *weights = Py_None;
+    PyArrayObject *arr_x = NULL, *arr_edges = NULL, *arr_w = NULL, *hist = NULL;
+    npy_intp *ihist = NULL, bins, len, i, ix;
+    double *x = NULL, *e = NULL, *w = NULL, *dhist = NULL, min, max, norm, tx;
+
+    static char *kwlist[] = {"x", "edges", "weights", NULL};
+
+    if (!PyArg_ParseTupleAndKeywords(args, kwds, "OO|O:_histogram_uniform",
+            kwlist, &obj_x, &obj_edges, &weights)) {
+        goto fail;
+    }
+
+    arr_x = (PyArrayObject *)PyArray_ContiguousFromAny(obj_x, NPY_DOUBLE, 1, 1);
+    if (arr_x == NULL) {
+        PyErr_SetString(PyExc_TypeError, "Couldn't parse the input array.");
+        goto fail;
+    }
+    len = (npy_intp)PyArray_DIM(arr_x, 0);
+
+    arr_edges = (PyArrayObject *)PyArray_ContiguousFromAny(obj_edges, NPY_DOUBLE, 1, 1);
+    if (arr_edges == NULL) {
+        PyErr_SetString(PyExc_TypeError, "Couldn't parse the input edges.");
+        goto fail;
+    }
+    bins = (npy_intp)(PyArray_DIM(arr_edges, 0) - 1);
+
+    /* Handle empty array */
+    if (len == 0) {
+        hist = (PyArrayObject *)PyArray_ZEROS(1, &bins, NPY_INTP, 0);
+        if (hist == NULL) {
+            goto fail;
+        }
+        Py_DECREF(arr_edges);
+        Py_DECREF(arr_x);
+        return (PyObject *)hist;
+    }
+
+    x = (double *)PyArray_DATA(arr_x);
+    e = (double *)PyArray_DATA(arr_edges);
+
+    min = e[0];
+    max = e[bins];
+
+    /* Possible divide-by-zero here */
+    norm = bins / (max - min);
```

`norm <= 0` should throw an error